src/Dashboard/App_Start/AppModule.cs

```diff
@@ -134,46 +133,37 @@ public static void Load(ContainerBuilder builder)
             }
         }
 
-        // Get a fast log table name 
+        // Get a fast log table name. Returns a table name prefix if 
         // OR return null to use traditional logging. 
-        private static CloudTable TryGetLogTable(CloudTableClient tableClient)
-        {
-            string logTableName = ConfigurationManager.AppSettings[FunctionLogTableAppSettingName];
-            if (string.IsNullOrWhiteSpace(logTableName))
+        private static IEpochTableProvider GetNewLoggerTableProvider(CloudTableClient tableClient)
+        {         
+            string logTablePrefix = ConfigurationManager.AppSettings[FunctionLogTableAppSettingName];
+            if (string.IsNullOrWhiteSpace(logTablePrefix))
             {
-                // Check for default name
-                string defaultName = LogFactory.DefaultLogTableName;
-                var table = tableClient.GetTableReference(defaultName);
-
                 var ver = ConfigurationManager.AppSettings[FunctionExtensionVersionAppSettingName];
-                if (!string.IsNullOrWhiteSpace(ver))
+                if (string.IsNullOrWhiteSpace(ver))
                 {
-                    if (string.Equals(ver, FunctionExtensionVersionDisabled, StringComparison.OrdinalIgnoreCase))
-                    {
-                        // Explicitly set to old mode. 
-                        return null;
-                    }
-                    else
-                    {
-                        // Appsetting specifically opts us in. Use fast-tables.
-                        table.CreateIfNotExists();
-                        return table;
-                    }
+                    // No Func appsetting, this is using sdk-style older logging. 
+                    return null;
                 }
 
-                if (table.Exists())
+                if (string.Equals(ver, FunctionExtensionVersionDisabled, StringComparison.OrdinalIgnoreCase))
```

Comment:
I prefer to separate the cases out - and then it gives me a place to comment each. 

---
In reply to: [82874256](https://github.com/Azure/azure-webjobs-sdk/pull/862#discussion_r82874256) [](ancestors = 82874256)